### PR TITLE
MULE-10548: Cleanup WorkManager and WorkListener

### PR DESCRIPTION
**Pull request by elrodro83**

### CODE REVIEW

File: core-tests/src/test/java/org/mule/runtime/core/routing/AsynchronousUntilSuccessfulProcessingStrategyTestCase.java [hunk 1]
```diff
@@ -120,6 +120,14 @@ public void setUp() throws Exception {
                                                                                                           .build());
   }
 
+  @After
+  public void after() {
```
Comment (dfeist) [inline core-tests/src/test/java/org/mule/runtime/core/routing/AsynchronousUntilSuccessfulProcessingStrategyTestCase.java]: Shouldn't abstract test case be doing this?

File: core-tests/src/test/java/org/mule/runtime/core/work/MuleEventWorkTestCase.java [hunk 2]
```diff
@@ -41,7 +41,7 @@ protected void doSetUp() throws Exception {
 
   @Test
   public void testScheduleMuleEventWork() throws Exception {
-    muleContext.getWorkManager().scheduleWork(new TestMuleEventWork(originalEvent));
+    muleContext.getSchedulerService().computationScheduler().submit(new TestMuleEventWork(originalEvent));
```
Comment (dfeist) [inline core-tests/src/test/java/org/mule/runtime/core/work/MuleEventWorkTestCase.java]: Why computation? Or is this not important, any pool is the same?

File: core/src/main/java/org/mule/runtime/core/connector/ConnectException.java [hunk 3]
```diff
@@ -67,7 +68,7 @@ private void readObject(ObjectInputStream in) throws IOException, ClassNotFoundE
     }
   }
 
-  public void handleReconnection() {
+  public void handleReconnection(Executor retryExecutor) {
```
Comment (dfeist) [inline core/src/main/java/org/mule/runtime/core/connector/ConnectException.java]: Why Executor and not ExecutorService?

File: core/src/main/java/org/mule/runtime/core/routing/AsynchronousUntilSuccessfulProcessingStrategy.java [hunk 4]
```diff
@@ -72,19 +69,13 @@ public void initialise() throws InitialisationException {
 
   @Override
   public void start() {
-    final String threadPrefix =
-        String.format("%s%s.%s", ThreadNameHelper.getPrefix(getUntilSuccessfulConfiguration().getMuleContext()),
-                      getUntilSuccessfulConfiguration().getFlowConstruct().getName(), "until-successful");
-    pool = getUntilSuccessfulConfiguration().getThreadingProfile().createPool(threadPrefix);
-    scheduledRetriesPool = getUntilSuccessfulConfiguration().createScheduledRetriesPool(threadPrefix);
+    pool = muleContext.getSchedulerService().ioScheduler();
```
Comment (dfeist) [inline core/src/main/java/org/mule/runtime/core/routing/AsynchronousUntilSuccessfulProcessingStrategy.java]: Would it be a good idea to allow a name to be passed when a pool is received, so any logging or error that occur can be associated back to where pool is used.  Or is there another mechanism to do this?

File: tests/integration/src/test/resources/until-successful-seconds-test.xml [hunk 5]
```diff
@@ -76,20 +76,4 @@
             </processor-chain>
         </until-successful>
     </flow>
-
-    <flow name="asynchronous-using-threading-profile">
```
Comment (dfeist) [inline tests/integration/src/test/resources/until-successful-seconds-test.xml]: Why is this removed?

File: tests/performance/src/test/resources/scatter-gather-perf-test.xml [hunk 6]
```diff
@@ -22,15 +22,6 @@
         </scatter-gather>
     </flow>
 
-    <flow name="sequentialProcessing">
```
Comment (dfeist) [inline tests/performance/src/test/resources/scatter-gather-perf-test.xml]: Isn't there another wy to do sequential processing now, to migrate this test to use?

File: core-tests/src/test/java/org/mule/runtime/core/routing/AsynchronousUntilSuccessfulProcessingStrategyTestCase.java (hunk 1 above)
Comment (elrodro83) [inline core-tests/src/test/java/org/mule/runtime/core/routing/AsynchronousUntilSuccessfulProcessingStrategyTestCase.java]: That's only the case for AbstractMuleContextTestCase. This text extends from AbstractMuleTestCase and creates the service within itself.

File: core-tests/src/test/java/org/mule/runtime/core/work/MuleEventWorkTestCase.java (hunk 2 above)
Comment (elrodro83) [inline core-tests/src/test/java/org/mule/runtime/core/work/MuleEventWorkTestCase.java]: Any pool is the same here, since just a test task is scheduled. This task just performs some assertions.

File: core/src/main/java/org/mule/runtime/core/routing/AsynchronousUntilSuccessfulProcessingStrategy.java (hunk 4 above)
Comment (elrodro83) [inline core/src/main/java/org/mule/runtime/core/routing/AsynchronousUntilSuccessfulProcessingStrategy.java]: No, that's a good idea actually. Created MULE-11018 for that

File: tests/performance/src/test/resources/scatter-gather-perf-test.xml (hunk 6 above)
Comment (elrodro83) [inline tests/performance/src/test/resources/scatter-gather-perf-test.xml]: no. Is that still a requirement for scatter gather?

Comment (dfeist) [inline tests/performance/src/test/resources/scatter-gather-perf-test.xml]: I'd check with Mariano G..
